### PR TITLE
Updating the function `compute_apparent_ABmag`

### PR DESCRIPTION
**Pull request by bazkiaei**
This function was one of the inconsistence functions in the code and the goal of this PR is to make it consistent with other functions #70 . Changing this function could cause some effects on other functions and create a need for change to some other functions as well.
The changes that are made here are inspired by the jupyter outflow that is provided in #71 and these changes will be applied to that PR finally.
This is one more step, maybe the most important one, to divide `prepare_mocks` and remove it from the code.

### CODE REVIEW

File: mocks/mocks.py
Comment (bazkiaei) [inline mocks/mocks.py:246]: With respect to the jupyter outflow, which is provided in #71 , there will be no need for this function anymore. But I will remove it and its test when this PR is approved.

File: mocks/tests/test_mocks.py
Comment (bazkiaei) [inline mocks/tests/test_mocks.py:168]: With respect to new changes that are made, inputs for this test should be changed. I prefer to make those changes after this PR is approved and in another PR.

**Comment by bazkiaei** [comment]
@AstroHuntsman/spiderlings 
Ready for review.

**Comment by lspitler** [comment]
@bazkiaei suggest renaming to just `compute_apparent_mag` and allow someone to send an argument `mag_units=u.ABmag` that defaults to AB mags.

**Review by fergusL — APPROVED** [review]
Looks good to me

**Comment by bazkiaei** [comment]
> Looks good to me

Thanks, I will wait for approval from @AstroHuntsman/arachne as well.

File: mocks/mocks.py
Comment (lspitler) [inline mocks/mocks.py:606]:
```suggestion
    absolute_mag =\
```

File: mocks/mocks.py
Comment (lspitler) [inline mocks/mocks.py:582]:
```suggestion
    environment with respect to a given distance.
```

File: mocks/mocks.py
Comment (lspitler) [inline mocks/mocks.py:588]:
```suggestion
    sim_particle_luminosities : astropy.units.Quantity
```

File: mocks/mocks.py
Comment (lspitler) [inline mocks/mocks.py:592]: Doesn't this come packed in the `config` dictionary? Suggest removing this.

**Review by lspitler — APPROVED** [review]

File: mocks/mocks.py
Comment (bazkiaei) [reply mocks/mocks.py:592]: No, it does not. But we should make a decision about how to introduce known cosmologies to the code.

**Pull request merged**
